tiledb/sm/array_schema/array_schema.cc

```diff
@@ -260,6 +260,10 @@ uint64_t ArraySchema::cell_size(const std::string& name) const {
     return constants::timestamp_size;
   }
 
+  if (name == constants::delete_condition_marker_hash) {
+    return sizeof(size_t);
```

Comment:
std::hash returns a size_t, does that mean it would return different values based on the platform it's compiled on?

Reply:
`size_t` is generally 4 bytes on 32-bit OSes (although it could theoretically be smaller).

Reply:
A deeper issue here is that `std::hash` is not guaranteed to be deterministic beyond the lifetime of the running process (not even identical invocations of the same binary), so it can not be used to generate durable keys [[1](https://en.cppreference.com/w/cpp/utility/hash)]:

> Hash functions are only required to produce the same result for the same input within a single execution of a program; this allows salted hashes that prevent collision denial-of-service attacks.

(IIUC from conversation, we are not yet storing these dcmh files, so this is not a critical issue in any merged code - but we'll have to switch to a fully deterministic hash before shipping)